### ISSUE TITLE
CLI - User is able to list all template versions including the ones not valid for that version of rancher

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#21733]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
**Setup:**
```
rancher version: v2.3.2(custom image for testing)
chart - fluentd-aggregator
chart version: latest - rancher_min_version: 2.3.2 and rancher_max_version: 2.4.0
chart version: v0.1.0 - rancher_max_version: 2.3.2
chart version: v0.2.1 - rancher_min_version: 2.3.3
```
1. Download rancher CLI 
2. Login to rancher and switch context to your project.

**Issue#1**
Command - `./rancher app show-template` for the fluentid app lists all template version including the ones not valid for that version of rancher.

```
Sowmyas-MacBook-Pro:rancher-v2.3.0-rc2 sowmyaviswanathan$ ./rancher app show-template p-wf2fl:b1-fluentd-aggregator
0.1.0
0.1.2
0.1.6
0.2.1
0.3.0
```

**Expected Result:**
Should only list the template versions valid for the rancher version - 
```
0.1.0
0.1.2
0.1.6
```
**Issue#2**
Not able to upgrade the app version to `0.1.6` which is listed as one of the available ones.
command used: `./rancher app upgrade a-cqrpt 0.1.6`
Error: `FATA[0000] version 0.1.6 is not valid `

```
Sowmyas-MacBook-Pro:rancher-v2.3.0-rc2 sowmyaviswanathan$ ./rancher app ls
ID                NAME      STATE     CATALOG      TEMPLATE             VERSION
p-wf2fl:a-cqrpt   a-cqrpt   active    p-wf2fl/b1   fluentd-aggregator   0.1.2
p-wf2fl:a-zgd2c   a-zgd2c   active    c-s84wq/b2   fluentd-aggregator   0.1.2
Sowmyas-MacBook-Pro:rancher-v2.3.0-rc2 sowmyaviswanathan$ ./rancher app upgrade a-cqrpt 0.1.6
FATA[0000] version 0.1.6 is not valid                   
Sowmyas-MacBook-Pro:rancher-v2.3.0-rc2 sowmyaviswanathan$ 
```

**Comment by dramich** [comment]
Available in v2.3.0-rc3

**Comment by sowmyav27** [comment]
verified with `v2.3.0-rc3`
```
rancher version - 2.3.2 (custom image)
chart - fluentd-aggregator
chart version: latest - rancher_min_version: 2.3.2 and rancher_max_version: 2.4.0
chart version: v0.1.0,0.1.2,0.1.6 - rancher_max_version: 2.3.2
chart version: v0.2.1 - rancher_min_version: 2.3.3
```
./rancher app show-template shows only template versions valid to the rancher version.

```
$ ./rancher app show-template c-npslj:sowmya-cluster-fluentd-aggregator
0.1.0
0.1.2
0.1.6
$ ./rancher app show-template p-dkzlv:sowmya-project-fluentd-aggregator
0.1.0
0.1.2
```
